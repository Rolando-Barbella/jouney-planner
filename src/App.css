.App {
  height: 100vh; 
  display: flex; 
  align-items: center; 
  justify-content: center;
  flex-direction: column;
}

h1, h2 {
  font-weight: normal;
}

ul {
  padding: 0;
  list-style-type: none;
}

li {
  width: 3.7em;
  height: 3.7em;
  text-align: center;
  line-height: 3.7em;
  border-radius: 1.9em;
  margin-bottom: 1.3em;
  color: white;
  position: relative;
  background: #bbddff;
  border: solid 1px #8abdf0;
}

.journey-planner-container { 
  height: 70vh; 
  display: flex; 
  justify-content: center;
  flex-direction: column;
  padding: 12px;
  width: 50%;
  box-shadow: 1px 6px 1px 0 rgba(0,0,0,.1);
  border-radius: 4px;
  border: solid 1px #f6f6f6;
}

.journey-planner-container h2 {
  font-weight: 400;
  text-align: center;
}

.wrapper-journey-planner {
  display: flex;
  justify-content: space-between;
  margin-left: 20px;
  margin-right: 20px;
}

.result-journey {
  display: inline-block;
  margin-right: 10px;
}

.wrapper-result-journey {
  display: flex;
  justify-content: center;
  flex-direction: column;
  align-items: center;
}

.wrapper-result-journey li::before{
  content: '';
  position: absolute;
  top: 1.65em;
  left: -1em;
  width: 5.5em;
  height: .2em;
  background: dodgerblue;
  z-index: -1;
}

.active-circle {
  background: dodgerblue;
  border: solid 1px #1873cc;
  font-weight: 500;
}

.button-container {
  display: flex;
  justify-content: center;
  align-items: center;
}

button{
  background-image: linear-gradient(to top left,#4ca64c,#008000);
  color: #fff !important;
  display: inline-block;
  font-family: inherit;
  font-weight: 700;
  border: 0;
  border-radius: 2rem;
  white-space: nowrap;
  padding: 1rem 1.5rem;
  line-height: 1.4;
  position: relative;
  text-align: center;
  -webkit-transition: 70ms;
  transition: 70ms;
  position: relative;
  cursor: pointer;
  font-size: 14px;
}

button:hover{
  background-image: linear-gradient(to top left,#008000, #4ca64c );
}
