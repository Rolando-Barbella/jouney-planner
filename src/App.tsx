import React, {  useState } from 'react'

import { dijkstra } from "./utils/shortest-journey";
import './App.css';

interface ISomeObject {
  start: Object;
  A: Object;
  B: Object;
  C: Object;
  D: Object;
  finish: Object;
}
// From A to D
const route: { [key: string]: number | object } = {
  start: { A: 1 },
  A: { C: 4, B: 2 },
  B: { A: 8, D: 1 },
  C: { D: 6, finish: 3 },
  D: { A: 4, finish: 1 },
  finish: {}
};

// console.log(dijkstra(route));

function App() {
  const startJourneis: Array<string> = ['A', 'B', 'C', 'D'];
  const endJourneis: Array<string> = ['E', 'F', 'G', 'J'];

  const [ journey, setJourney ] = useState({ start: '', end: '' });
  // const [ startJourney, setStartJourney ] = useState(null);
  // const [ atoBState, setatoBState ] =  useState(route);
  return (
    <div className="App">
      <h2>Start</h2>
      <section className="journey-container">
        <div data-testid="start-options">
          {
            startJourneis.map(start => (
              <p
                key={start}
                className={journey.start === start ? "active-circle" : ""}
                onClick={() => setJourney({...journey, start })}
              >
                {start}
              </p>
            ))
          }
        </div>
        <div data-testid="end-options">
          {
            endJourneis.map(end => (
              <p
                key={end}
                className={journey.end === end ? "active-circle" : ""}
                onClick={() => setJourney({...journey, end })}
              >
                {end}
              </p>
            ))
          }
        </div>
      </section>
    </div>
  );
}

export default App;
