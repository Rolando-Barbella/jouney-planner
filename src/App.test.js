import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import App from './App';

afterEach(cleanup);

it('renders without crashing', () => {
  const { asFragment } = render(<App />);

  expect(asFragment()).toMatchSnapshot();
});

it('should display start title', () => {
  const startTitle = 'Start';
  const { getByText } = render(
    <App>{startTitle}</App>
  );

  expect(getByText(startTitle)).toBeInTheDocument()
});

it('it displays start options', () => {
  const { getByTestId } = render(<App />);
  const startOptions = getByTestId('start-options');

  expect(startOptions.children.length).toBe(4);
});

it('displays end options', () => {
  const { getByTestId } = render(<App/>);
  const endOptions = getByTestId('end-options');

  expect(endOptions.children.length).toBe(4);
});
