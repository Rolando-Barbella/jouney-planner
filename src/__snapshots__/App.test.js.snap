// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <h2>
      Start
    </h2>
    <section
      class="journey-container"
    >
      <div
        data-testid="start-options"
      >
        <p
          class=""
        >
          A
        </p>
        <p
          class=""
        >
          B
        </p>
        <p
          class=""
        >
          C
        </p>
        <p
          class=""
        >
          D
        </p>
      </div>
      <div
        data-testid="end-options"
      >
        <p
          class=""
        >
          E
        </p>
        <p
          class=""
        >
          F
        </p>
        <p
          class=""
        >
          G
        </p>
        <p
          class=""
        >
          J
        </p>
      </div>
    </section>
  </div>
</DocumentFragment>
`;
